src/ClientData/ThreadTrackPaneData.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <ClientData/ThreadTrackPaneData.h>
+
+namespace orbit_client_data {
+
+size_t ThreadTrackPaneData::GetNumberOfTimers() const {
+  absl::MutexLock lock(&scope_tree_mutex_);
+  // Special case. ScopeTree has a root node in depth 0 which shouldn't be considered.
+  return scope_tree_.Size() - 1;
+}
+
+uint32_t ThreadTrackPaneData::GetMaxDepth() const {
+  absl::MutexLock lock(&scope_tree_mutex_);
+  return scope_tree_.Height() - 1;
+}
+
+const orbit_client_protos::TimerInfo& ThreadTrackPaneData::AddTimer(
+    orbit_client_protos::TimerInfo timer_info) {
+  const auto& timer_info_ref = track_pane_data_.AddTimer(/*depth=*/0, timer_info);
+  UpdateMinTime(timer_info.start());
+  UpdateMaxTime(timer_info.end());
```

Comment:
as far as I can see we have exactly the same behavior in track_pane_data_.AddTimer() why not just delegate this to track_pane_data_?